python-squid/src/test/java/org/sonar/python/parser/python_v3/PythonV3Test.java

```diff
@@ -46,6 +46,21 @@ public void function_declaration(){
   }
 
   @Test
+  public void asyncAndAwait() {
+    setRootRule(PythonGrammar.FUNCDEF);
+    assertThat(p).matches("async def fun(): pass");
+
+    setRootRule(PythonGrammar.ASYNC_STMT);
+    assertThat(p).matches("async with x.bar() as foo: pass");
+    assertThat(p).matches("async for var in x: pass");
+
+    setRootRule(PythonGrammar.STATEMENT);
+    assertThat(p).matches("async = 2");
+    assertThat(p).matches("await = 2");
+    assertThat(p).matches("await async_func");
```

Comment:
@ivandalbosco actually `await foo()` is expression. It's better to test it with root rule `TEST` or `EXPR`